src/Runner.Worker/JobExtension.cs

```diff
@@ -74,6 +74,10 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                         {
                             // print out HostName for self-hosted runner
                             context.Output($"Runner name: '{setting.AgentName}'");
+                            if (message.Variables.TryGetValue("system.runnerGroupName", out VariableValue runnerGroupName))
+                            {
+                              context.Output($"Runner group name: '{runnerGroupName}'");
```

Comment:
i think you might need to do:

```csharp
context.Output($"Runner group name: '{runnerGroupName.Value}'");
```

otherwise i think it will always output `Runner group name: 'Object'` since VariableValue does not override ToString()

Reply:
Good spot!